### DISCUSSION TITLE
section selector with level specifier?

### DISCUSSION THREAD

**Discussion by yshavit** [yshavit/mdq#120] [Q&A] [answered]
Right now, section selectors don't differentiate between the level of header:

```markdown
# foo

☝️ this...

## foo

☝️ ... matches the same as this.
```

Do we want to add an option to specify the level?

**Comment by yshavit** [comment]
### Assuming yes, what would the syntax be?

Suggestion:

```
{1}# foo
```

This would match an `h1` heading: `# foo` but not `## foo`.

Alternatively, we could do:

```
#{1} foo
```

This is clearer, but I'd need to think how it would interact with the possible quoting options being discussed in #56 (e.g. `{"EOF"}text text textEOF`).

Regardless, we could also add `>`, `<`, `>=`, `<=` in front of the number:

```
{>=2}# foo
```

... would match `## foo` and `### foo` but not `# foo`.

**Reply by danielwerg** [reply]
What about regex style?

`#{1}`: h1
`#{2,3}`: h2, h3
`#{2,}`: h2, h3, h4, h5, h6 (h7? :D)

**Reply by yshavit** [reply]
Oh, I like that!

**Comment by PhilipHayes** [comment]
Did we ever get around to this?

**Reply by yshavit** [reply]
Thanks for the nudge! I had a bunch on my plate, and this fell off my radar.

I have a PR up for it now: #419. Take a look (particularly at the description, and/or integ tests) and tell me what you think!

**Reply by PhilipHayes** [reply]
yeah that's perfect!

I was going to ask about getting only the headers but I think this would do it:

on Mac for example
`find doc/* -name '*.md' -exec mdq "# " {} \; | grep '^#'` <- gets ALL headers at All levels
`find doc/* -name '*.md' -exec mdq "# " {} \; | grep '^# '` <- gets all h1's 
but it's raw strings which is fine. The json output is incredibly helpful. and combined with jq, there's recipe potential.

**Comment by yshavit** [comment, marked as answer]
This is now possible as of #419, released in v0.10.0.

Syntax:

```text
#{2} "matches exactly h2 sections"
#{2,} "matches h2 - h6 sections"
#{,2} "matches h1 - h2 sections"
#{2,4} "matches h2 - h4 sections"
```
